tests/test_type/test_conversions.py

```diff
@@ -105,3 +108,33 @@ def test_tuple():
      assert isinstance(tuple_(sint(1,4)), TupleType)
      assert isinstance(tuple_(uint(1,4)), TupleType)
 
+
+@pytest.mark.parametrize("type_,value",
+                         product([bits, sint, uint], [-5, 0, 10]))
+def test_zext(type_, value):
+    if type_ != sint:
+        value = abs(value)
+    in_ = type_(value, 16)
+    # TODO(rsetaluri): Ideally, zext(bits) should return an object of type
+    # BitsType, instead it returns an object of type ArrayType. For now, we wrap
+    # the result of zext() in bits().
+    out = type_(zext(in_, 16))
+    assert len(out.bits()) == 32
+    # If we have a negative number, then zext should not return the same (signed
+    # value). It will instead return the unsigned interpretation of the original
+    # bits.
+    if value < 0:
+        assert int(out) == seq2int(in_.bits())
+    else:
+        assert int(out) == value
+
+
+@pytest.mark.parametrize("value", [-5, 0, 10])
+def test_sext(value):
+    in_ = sint(value, 16)
+    # TODO(rsetaluri): Ideally, zext(sint) should return an object of type
+    # SintType, instead it returns an object of type ArrayType. For now, we wrap
+    # the result of zext() in sint().
```

Comment:
Can we open an issue on this? Basically the extension operators should preserve the input type.